File: 0029-divide-two-integers/0029-divide-two-integers.py
class Solution:
    def divide(self, dividend: int, divisor: int) -> int:
        if((int(dividend/divisor) > 2**31 - 1)):
            return 2**31 - 1
        elif(int(dividend/divisor) < -2**31):
            return -2**31    
        elif(dividend>=0 and divisor>=0):
            return dividend//divisor
        elif(dividend<0 and divisor<0):
            return int(dividend/divisor)    
        elif(dividend<0 or divisor<0):
            return -int(abs(dividend)/abs(divisor))

File: 0029-divide-two-integers/0029-divide-two-integers.cpp
class Solution {
public:
    int divide(int dividend, int divisor) {
        long long ld_dividend = dividend;
        long long ld_divisor = divisor;
        long long result = ld_dividend / ld_divisor;
        
        if (result > (1LL<<31)-1) {
            return 2147483647;
        }
        if (result < -(1LL<<31)) {
            return -2147483648;
        }
        return (int)result;
    }
};